### PR TITLE
Use `pathy-argonaut` and remove duplicate code in `quasar`

### PR DESCRIPTION
**Pull request by jedesah**

### CODE REVIEW

File: core/src/main/scala/quasar/fs/PathyCodecJson.scala [hunk 1]
```diff
@@ -21,19 +21,12 @@ import quasar.Predef._
 import argonaut._, Argonaut._
 import pathy.Path, Path._
 
-// TODO: These belong in argonaut sub-module in scala-pathy
 object PathyCodecJson {
```
Comment (wemrysi) [inline core/src/main/scala/quasar/fs/PathyCodecJson.scala]: What do you think about rename this object to `APath` now?

**Comment by wemrysi** [comment]
One minor comment, otherwise LGTM :+1:

File: core/src/main/scala/quasar/fs/PathyCodecJson.scala (hunk 1 above)
Comment (jedesah) [inline core/src/main/scala/quasar/fs/PathyCodecJson.scala]: Actually, how about just sticking it in `fs/package.scala` right next to the definition of `APath`?

Comment (wemrysi) [inline core/src/main/scala/quasar/fs/PathyCodecJson.scala]: I'd still prefer the `APath` object, regardless, for organizational reasons and for making it easier to see where the instances come from when looking at imports and if you want to move the object def to the package object, I won't object.

**Comment by jedesah** [comment]
Didn't really mean to merge this as two commits, but I guess it's not too bad.

**Pull request closed without merging**